pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -281,11 +293,39 @@ func manageRouteControllerManagerClientCA_v311_00_to_latest(client coreclientv1.
 	return resourceapply.SyncConfigMap(context.Background(), client, recorder, util.KubeAPIServerNamespace, apiserverClientCA, util.RouteControllerTargetNamespace, apiserverClientCA, []metav1.OwnerReference{})
 }
 
+// disableControllers disables controllers based on capabilities.
+// By default, all controllers are enabled. To disable specific controllers,
+// '-foo' disables the controller named 'foo'. For example, to disable the
+// OpenShiftServiceAccountController, one would use "-openshift.io/serviceaccount".
+func disableControllers(clusterVersion *configv1.ClusterVersion) []string {
+	controllers := []string{"*"}
+	knownCaps := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.KnownCapabilities...)
+	capsEnabled := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.EnabledCapabilities...)
+
+	for cont, cap := range controllerCapabilities {
+		if knownCaps.Has(cap) && !capsEnabled.Has(cap) {
+			controllers = append(controllers, fmt.Sprintf("-%s", cont))
+		}
+	}
+	return controllers
+}
+
 // similar logic for route-controller-manager in manageRouteControllerManagerConfigMap_v311_00_to_latest
-func manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(kubeClient kubernetes.Interface, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorConfig *operatorapiv1.OpenShiftControllerManager) (*corev1.ConfigMap, bool, error) {
+func manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(clusterVersionLister configlisterv1.ClusterVersionLister, clusterVersionName string, kubeClient kubernetes.Interface, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorConfig *operatorapiv1.OpenShiftControllerManager) (*corev1.ConfigMap, bool, error) {
```

Comment:
`clusterVersionName` argument is no longer needed after the last refactoring. `clusterVersionLister.Get("version")` will do as well.

Reply:
done